Client/Application/LeanplumIntegration.swift

```diff
@@ -0,0 +1,201 @@
+private let LeanplumAppIdKey = "LeanplumAppId"
+private let LeanplumKeyKey = "LeanplumKey"
+
+private let log = Logger.browserLogger
+
+private enum LeanplumEnvironment: String {
+    case development = "development"
+    case production = "production"
+}
+
+enum LeanplumEventName: String {
+    case firstRun = "First Run"
+    case secondRun = "Second Run"
+    case openedApp = "Opened App"
+    case openedLogins = "Opened Login Manager"
+    case openedBookmark = "Opened Bookmark"
+    case openedNewTab = "Opened New Tab"
+    case interactWithURLBar = "Interact With Search URL Area"
+    case savedBookmark = "Saved Bookmark"
+    case openedTelephoneLink = "Opened Telephone Link"
+    case openedMailtoLink = "Opened Mailto Link"
+    case downloadedImage = "Download Media - Saved Image"
+    case savedLoginAndPassword = "Saved Login and Password"
+}
+
+private enum SupportedLocales: String {
+    case US = "en_US"
+    case DE = "de"
+    case UK = "en_GB"
+    case CA_EN = "en_CA"
+    case AU = "en_AU"
+    case TW = "zh_TW"
+    case HK = "en_HK"
+    case SG_EN = "en_SG"
+}
+
+private struct LeanplumSettings {
+    var environment: LeanplumEnvironment
+    var appId: String
+    var key: String
+}
+
+class LeanplumIntegration {
+    static let sharedInstance = LeanplumIntegration()
+
+    // Setup
+
+    fileprivate weak var profile: Profile?
+    private var enabled: Bool = false
+    
+    func shouldSendToLP() -> Bool {
+        return enabled && Leanplum.hasStarted()
+    }
+
+    func setup(profile: Profile) {
+        self.profile = profile
+    }
+
+    fileprivate func start() {
+        self.enabled = self.profile?.prefs.boolForKey("settings.sendUsageData") ?? true
+        guard self.enabled else {
+            return
+        }
+        
+        guard SupportedLocales(rawValue: Locale.current.identifier) != nil else {
+            return
+        }
+
+        if Leanplum.hasStarted() {
+            log.error("LeanplumIntegration - Already initialized")
+            return
+        }
+
+        guard let settings = getSettings() else {
+            log.error("LeanplumIntegration - Could not load settings from Info.plist")
+            return
+        }
+
+        switch settings.environment {
+        case .development:
+            log.info("LeanplumIntegration - Setting up for Development")
+            Leanplum.setDeviceId(UIDevice.current.identifierForVendor?.uuidString)
+            Leanplum.setAppId(settings.appId, withDevelopmentKey: settings.key)
+        case .production:
+            log.info("LeanplumIntegration - Setting up for Production")
+            Leanplum.setAppId(settings.appId, withProductionKey: settings.key)
+        }
+        Leanplum.syncResourcesAsync(true)
+        setupTemplateDictionary()
+
+        var userAttributesDict = [AnyHashable: Any]()
+        userAttributesDict["Alternate Mail Client Installed"] = "mailto:"
+        userAttributesDict["Focus Installed"] = canInstallFocus()
```

Comment:
I think your first concern is a minor, so I'll punt on it for now. But your second point is right - when @bkmunar refactored this per @jhugman review, he got the logic flipped (it happens :) ).